iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPSubsystemAdd.java

```diff
@@ -180,9 +185,11 @@ public void execute(DeploymentProcessorTarget processorTarget) {
         final boolean sslConfigured = this.setupSSLFactories(props);
 
         // create the service that initializes and starts the CORBA ORB.
-        CorbaORBService orbService = new CorbaORBService(props);
-        final ServiceBuilder<ORB> builder = context.getServiceTarget().addService(CorbaORBService.SERVICE_NAME, orbService);
-        org.jboss.as.server.Services.addServerExecutorDependency(builder, orbService.getExecutorInjector());
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(org.wildfly.iiop.openjdk.IIOPRootDefinition.IIOP_CAPABILITY);
+        final Consumer<CorbaORBService> serviceConsumer = builder.provides(org.wildfly.iiop.openjdk.IIOPRootDefinition.IIOP_CAPABILITY);
```

Comment:
The RuntimeCapability definition specifies a service type of Void, but here, you have this service providing a CorbaOrbService.
Perhaps this should provide an `org.omg.CORBA.ORB`?

Reply:
This is correct type, the expected consumer type is defined in `CorbaORBService`